xbmc/pvr/addons/PVRClient.h

```diff
@@ -570,10 +570,12 @@ class CPVRClient : public ADDON::IAddonInstanceHandler
    * @brief Fill the given container with the properties required for playback of the given channel. Values are obtained from the PVR backend.
    * @param channel The channel.
    * @param props The container to be filled with the stream properties.
+   * @param fromEpgAsLive True if this call resulted from PVR_STREAM_PROPERTY_EPGPLAYBACKASLIVE being set from GetEPGTagStreamProperties(), false otherwise.
    * @return PVR_ERROR_NO_ERROR on success, respective error code otherwise.
    */
   PVR_ERROR GetChannelStreamProperties(const std::shared_ptr<const CPVRChannel>& channel,
-                                       CPVRStreamProperties& props) const;
+                                       CPVRStreamProperties& props,
+                                       bool fromEpgAsLive) const;
```

Comment:
Do you see a way to avoid the bool param, for example by providing an new method? I'm not a fan of bool parameters, because resulting calling code is (almost ever) not really readable.

Reply:
We could go down the route of defining an enum to use in the call, that would improve readability. Just need to make sure it works through the  c part of the API call.

I don't think a new method is a good idea here as it's a callback, as then the add-ons would need to handle multiple entry points and will mean they just all create the method with the signature above to bring it back together.

Reply:
> Just need to make sure it works through the c part of the API call.

I'm not against a bool param at C-API level, I'm only concerned about C++.

> I don't think a new method is a good idea here as it's a callback

`GetChannelStreamProperties` is not a callback as far as I know and I still think at C++ level it is just fine to have two methods. But ofc up to you to go with an enum class if you prefer. ;-)